### PR TITLE
use more verbose method for attribute checking

### PR DESCRIPTION
**Pull request by cthayer**
The less verbose method is not robust across versions of chef

### CODE REVIEW

**Comment by mitch-roblox** [comment]
Out of curiosity, what versions of Chef does it fail on?

**Comment by cthayer** [comment]
It actually fails in all of them.  Seems that Chef does like doing `.dig()` on the `node` object.

Accessing attributes with `.` fails on Chef 13.

Using `node&` failed in a bunch of various scenarios on Chef 12 too.

I used `chef-shell` to test various syntaxes

**Review by warbirdcharles — APPROVED** [review]

**Comment by mitch-roblox** [comment]
Ahh, it fails because the node object doesn't inherit from `Mash`. It does work with sub-attributes, though, so this should work:

```
if node['cloud'] && node['cloud'].dig('provider')&.downcase == 'ec2'
```

`dig()` is really useful for deeply nested attributes.

Or you can use chef-sugar and `deep_fetch()`.

File: recipes/default.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@
 nslcd_enable = false
 case node['platform']
 when 'redhat', 'centos', 'scientific'
-  if node.dig('cloud', 'provider')&.downcase == 'ec2'
+  if node.key?('cloud') && node['cloud'].key?('provider') && node['cloud']['provider'].downcase == 'ec2'
```
Comment (mitch-roblox) [inline recipes/default.rb]: You can use the snippet on `dig()` I gave in my comment if you like, or not.

However, one thing I would suggest changing is avoiding `key?()`. This works equally well and is more succinct:

```
node['cloud'] && node['cloud']['provider'] && node['cloud']['provider'].downcase == 'ec2'
```

That seems to be the more idiomatic way for checking if an attr exists in Chef.

Comment (cthayer) [reply recipes/default.rb]: I used the `dig()` from your other comment.  It worked in `chef-shell`

**Pull request merged**